### PR TITLE
Floater refactoring

### PR DESCRIPTION
**Pull request by simonschaufi**
Load the data from an array instead of assigning every single value to the view.

### CODE REVIEW

**Comment by kevinpapst** [comment]
What is the reason behind this PR? I don't like rewriting code that works when there is no real benefit from that rewrite afterwards. To often something breaks ... and thats not related to your work but a general rule of thumb that proves to be true every day in every team^^

**Comment by simonschaufi** [comment]
I extended the table and don't need to assign every single variable to the view any more. It will be there automatically since the whole record is now available. this is good practice generally.

File: extensions/ki_expenses/floaters.php [hunk 1]
```diff
@@ -62,19 +53,26 @@
                 $view->projects[$projectData['projectID']] = $customerData['name'] . ':' . $projectData['name'];
             }
         } else {
-            $view->assign('id', 0);
-            $view->assign('edit_day', date("d.m.Y"));
-            $view->assign('edit_time', date("H:i:s"));
-            $view->assign('multiplier', '1' . $kga['conf']['decimalSeparator'] . '0');
+            // defaults
+            $expense = [
```
Comment (kevinpapst) [inline extensions/ki_expenses/floaters.php:57]: what about: move the defaults above if($id) in line 40 and then do a `$expense = array_merge($expense, get_expense($id))`?

File: extensions/ki_expenses/templates/scripts/floaters/add_edit_record.php [hunk 2]
```diff
@@ -35,67 +35,67 @@
...
-                            ?>
-                            <br/>
+                            ?><br/>
                             <input type="text" style="width:395px;margin-top:3px" tabindex="2" size="10" name="filter" id="filter" onkeyup="filter_selects('add_edit_expense_project_ID', this.value);"/>
                         </div>
                     </li>
                     <li>
                         <label for="edit_day"><?php echo $this->translate('day') ?>:</label>
-                        <input id='edit_day' type='text' name='edit_day' value='<?php echo $this->escape($this->edit_day) ?>' maxlength="10" size="10" tabindex="5" <?php if ($autoSelection): ?> onclick="this.select();" <?php endif; ?> />
+                        <input type="text" name="edit_day" id="edit_day" value='<?php echo $this->escape(date('d.m.Y', $this->expense['timestamp'])) ?>' maxlength="10" size="10" tabindex="5" <?php if ($autoSelection): ?> onclick="this.select();" <?php endif; ?> />
```
Comment (kevinpapst) [inline extensions/ki_expenses/templates/scripts/floaters/add_edit_record.php]: can you re-use one of the existing date / time formats?

**Comment by kevinpapst** [comment]
> this is good practice generally

says who?

File: extensions/ki_expenses/floaters.php (hunk 1 above)
Comment (simonschaufi) [reply extensions/ki_expenses/floaters.php:57]: I thought about it for a while in the beginning but then had the idea about completely separating the code into a "create" and an "update" Action. This is my end goal to make it even cleaner. I really don't like the mix of both together in one method.

File: extensions/ki_expenses/templates/scripts/floaters/add_edit_record.php (hunk 2 above)
Comment (simonschaufi) [reply extensions/ki_expenses/templates/scripts/floaters/add_edit_record.php]: sure!

**Comment by simonschaufi** [comment]
The controller should not care about the view. The view should be able to display all properties of an object (array in our case).

**Comment by kevinpapst** [comment]
Good argument!

**Review by kevinpapst — APPROVED** [review]

**Pull request merged**